src/main/java/jewellery/inventory/service/SaleService.java

```diff
@@ -51,13 +51,25 @@ public SaleResponseDto createSale(SaleRequestDto saleRequestDto) {
     throwExceptionIfProductIsPartOfAnotherProduct(sale.getProducts());
 
     Sale createdSale = saleRepository.save(sale);
+    setProductPriceDiscountSalePriceAndSale(createdSale);
     updateProductOwnersAndSale(sale.getProducts(), saleRequestDto.getBuyerId(), createdSale);
     logger.info("Sale created successfully. Sale ID: {}", createdSale.getId());
     return saleMapper.mapEntityToResponseDto(createdSale);
   }
 
-  private void throwExceptionIfSellerNotProductOwner(List<Product> products, UUID sellerId) {
-    for (Product product : products) {
+  private void setProductPriceDiscountSalePriceAndSale(Sale sale) {
+    for (int i = 0; i < sale.getProducts().size(); i++) {
+      sale.getProducts()
+          .get(i)
+          .setSalePrice(productService.getProductSalePrice(sale.getProducts().get(i).getProduct()));
+      sale.getProducts().get(i).setSale(sale);
```

Comment:
This method would look a bit better if we use sale.getProducts().forEach(....). Or alternatively, extract sale.getProducts().get(i) as currentProductPriceDiscount to not have so many repetitions of sale.getProducts().get(i)